### ISSUE TITLE
Octane - Laravel Request Class prepareForValidation not reflected in $request->get

### ISSUE THREAD

**Issue by yusufalper** [laravel/framework#44020]
- Laravel Version: 9.26.1
- PHP Version: 8.1.9
- Database Driver & Version: pgsql 14.5

### Description:
In a API endpoint taking "emails" as an array, I am validating values with a Request Class. In this Request class' prepareForValidation method:

`$this->merge([
    'emails' => array_unique($this->emails)
]);`

i have above lines. And a controller like this:

`foreach ($request->get('emails') as $email) {
    //process
}`

In local environment (without octane), I have no error or bug. It works expected. I am giving "aaa@bbb.com" twice, and it works only once. But in server (with octane, no other changes), when I give "aaa@bbb.com" twice again, it works twice.

I have finally solved the problem with using:

`$request->input()`

instead of 

`$request->get()` 

But I am curious, why $request->get() works fine in local but causing problems in server? Is it related to Octane? Has anyone encountered such a problem before?

**Comment by driesvints** [comment]
Hey there,

Can you first please try one of the support channels below? If you can actually identify this as a bug, feel free to open up a new issue with a link to the original one and we'll gladly help you out.

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
- [Discord](https://discordapp.com/invite/KxwQuKb)
- [Larachat](https://larachat.co)
- [IRC](https://web.libera.chat/?nick=laravelnewbie&channels=#laravel)

Thanks!
